### DISCUSSION TITLE
How to monitor healtchecks for issues

### DISCUSSION THREAD

**Discussion by pagdot** [healthchecks/healthchecks#698] [Q&A] [answered]
I had a recent outage of my selfhosted healthchecks instance. Now I want to monitor it's uptime to be able to be notified if another issue occurs.

I've also got uptime-kuma setup and may use it for monitoring.
I have a reverse proxy (swag by linuxserver), which shows a generic page if a service is down. So just checking for http 200 isn't enough.
uptime-kuma supports keywords, but I still need to figure them out.

Another option would be an endpoint provided by healthchecks which provides a status of the healthchecks instance

How do you all handle this?

**Comment by RealOrangeOne** [comment]
> which shows a generic page if a service is down. So just checking for http 200 isn't enough.

This is a problem. If a reverse proxy can't reach its upstream, it should [http 503](https://http.cat/503) or [http 502](https://http.cat/502). Without that assumption, it becomes basically impossible to detect a fail unless you're checking what the content of the response is.

A custom endpoint which looks are more integrated metrics of the application and responds based on that could be quite useful. I've built a few in my time using [`django-health-check`](https://github.com/KristianOellegaard/django-health-check).

**Comment by cuu508** [comment, marked as answer]
Yep, the proxy should not report 200 if the upstream is down.

Healthchecks has an `/api/v1/status/` endpoint. It runs a `SELECT 1` query and returns a HTTP 200 with "OK" in the response body if the query succeeds. IOW, if the web app can talk to the database, the status endpoint reports success.

https://github.com/healthchecks/healthchecks/blob/914202e45bfe29ce24f55125d7fe786490345f4a/hc/api/views.py#L569
